src/library/autodj/autodjprocessor.cpp

```diff
@@ -859,17 +862,29 @@ void AutoDJProcessor::playerPositionChanged(DeckAttributes* pAttributes,
             double transitionProgress = (thisPlayPosition - thisDeck->fadeBeginPos) /
                     (thisDeck->fadeEndPos - thisDeck->fadeBeginPos);
             double transitionStep = transitionProgress - m_transitionProgress;
+
+            if ((transitionProgress == 0.0)  && 
+                (m_crossfaderStartCenter))
+                {
+                setCrossfader(0.0); // So we start at the mid point before we make any fader moves.
+                }
             if (transitionStep > 0.0) {
                 // We have made progress.
                 // Backward seeks pause the transitions; forward seeks speed up
                 // the transitions. If there has been a seek beyond endPos, end
                 // the transition immediately.
-                double remainingCrossfader = crossfaderTarget - currentCrossfader;
-                double adjustment = remainingCrossfader /
-                        (1.0 - m_transitionProgress) * transitionStep;
-                // we move the crossfader linearly with
-                // movements in this track's play position.
-                setCrossfader(currentCrossfader + adjustment);
+                if ((m_crossfaderStartCenter) &&
+                    (transitionProgress <= ((thisDeck->fadeEndPos - thisDeck->fadeBeginPos)/2))) {
```

Comment:
If you mean to use the standard lib calls instead of the arithmetic, then sure, that makes the intent clearer.  (My C++ usage has been mostly on micro-controllers where one avoids both the speed and space overhead of libraries if possible.)